src/idc.py

```diff
@@ -0,0 +1,399 @@
+get_wide_dword = ida_bytes.get_wide_dword
+get_wide_byte = ida_bytes.get_wide_byte
+
+o_reg = 1
+o_mem = 2
+
+
+def get_program_context():
+    """
+    Get reusable program context: FlatProgramAPI and minAddress.
+
+    @return: (program_api, minAddress)
+    """
+    program_api = FlatProgramAPI(cp.currentProgram)
+    min_address = cp.currentProgram.getMinAddress().getOffset()
+    return program_api, min_address
+
+
+def get_memory_block_at_address(ea):
+    """
+    Get the memory block containing the given EA.
+
+    @param ea: IDA-style offset (relative to minAddress)
+    @return: MemoryBlock object or None if not found
+    """
+    program_api, min_address = get_program_context()
+    addr = program_api.toAddr(ea + min_address)
+    return cp.currentProgram.getMemory().getBlock(addr)
+
+
+def get_segm_attr(ea, attr):
+    """
+    Get segment attribute value by EA.
+
+    @param ea: any address in the segment (IDA-style EA)
+    @param attr: attribute (currently only SEGATTR_END supported)
+    @return: attribute value (IDA-style EA), or BADADDR if invalid
+    """
+    block = get_memory_block_at_address(ea)
+    if block is None:
+        return BADADDR
+
+    _, min_address = get_program_context()
+
+    if attr == SEGATTR_END:
+        return block.getEnd().getOffset() - min_address
+
+    print(f"[WARN] Unsupported segment attribute: {attr}")
+    return BADADDR
+
+
+def GetString(address, length):
+    """
+    was ported to ida_bytes get_strlit_contents according to IDA
+    address: linear address of the string
+    length: length of the string in bytes including  null terminator
+    return: a bytes-filled str object.
+    """
+    res = b""
+    program_api = FlatProgramAPI(cp.currentProgram)
+    try:
+        res = program_api.getBytes(program_api.toAddr(address), length + 1)
+        res = res.tolist()
+        res = [i if i >= 0 else (256 + i) for i in res]
+        res = array("B", res).tobytes()
+    except Exception as ex:
+        print("GetString failed: ", str(ex))
+    finally:
+        return res
+
+
+def get_segm_name(ea):
+    """
+    Get name of a segment
+
+    @param ea: any address in the segment
+
+    @return: segment name
+             "" - no segment at the specified address
+    """
+    block = get_memory_block_at_address(ea)
+    if block is None:
+        return ""
+
+    name = block.getName()
+    if name == "EXTERNAL":
+        return "extern"
+
+    return name
+
+
+def get_func_name(ea):
+    """
+    Retrieve function name
+
+    @param ea: any address belonging to the function
+    @return: null string - function doesn't exist
+             otherwise returns function name
+    """
+    program_api, min_address = get_program_context()
+    listing = cp.currentProgram.getListing()
+    func = listing.getFunctionAt(program_api.toAddr(min_address + ea))
+
+    if func is None:
+        return ""
+
+    return func.getName()
+
+
+def auto_wait():
+    pass
+
+
+def import_type(idx, type_name):
```

Comment:
DANGER: this function is platform (UEFI)-specific, therefore should be in a plugin, not in the translation layer that should be platform agnostic (e.g. ELF, PE for windows, etc.).

Try to remove it from here and place it in a plugin or, if not possible for now, in a temporary platform-specific file.